lab_4_summarization_textrank/README.md

```diff
@@ -0,0 +1,758 @@
+                     idf_values: dict[str, float]):
+        pass
+```
+
+У каждого экземпляра `Buddy` должны быть как минимум следующие атрибуты:
+* `_stop_words` - список стоп слов
+* `_punctuation` - список пунктуационных знаков
+* `_idf_values` - предобученные значения IDF для выделения ключевых слов с помощью `TFIDFAdapter`
+* `_text_preprocessor` - экземпляр класса `TextPreprocessor`, необходимой для предобработки текста
+* `_sentence_encoder` - экземпляр класса `SentenceEncoder`, необходимой для кодирования предложений
+* `_sentence_preprocessor` - экземпляр класса `SentencePreprocessor`, необходимой для предобработки предложений
+* `_paths_to_texts` - путь до текстов, которые будут составлять базу знаний помощника 
+* `_knowledge_database` - база знаний помощника со следующей структурой:
+  * Ключи - путь до текста
+  * Значения - словари со следующей структурой: `{'sentences': sentences, 'keywords': keywords, 'summary': summary}`
+    * `sentences` - предобработанные и закодированные предложения (каждый элемент - экземпляр `Sentence`) 
+    * `keywords` - ключевые слова для данного текста (кортеж из строк)
+    * `summary` - краткое содержание текста
+
+Значение по умолчанию для `_knowledge_database` - пустой словарь.
+
+При создании экземпляра класса `Buddy` база знаний должна заполняться информацией на основе текстов, переданных в аргументе `paths_to_texts`.
+
+#### Шаг 9.1 Заполнение базы знаний помощника
+
+Для заполнения базы знаний Вам необходимо реализовать метод `add_text_to_database`.
+
+Метод принимает на вход путь до текста.
+
+Метод:
+1. Считывает текст
+2. Извлекает из текста предложения
+   1. Предобрабатывает предложения
+   2. Кодирует предложения
+3. Извлекает из текста ключевые слова с помощью `TFIDFAdapter`
+4. Создаёт краткое содержание для текста с помощью `TextRankSummarizer`
+5. Записывает в переменную `_knowledge_database` пару ключ-значение, где ключ - уникальный идентификатор текста (путь до него); 
+значение - словарь со структурой, описанной ранее:
+    ```py
+    self._knowledge_database[text_id] = {
+        'sentences': sentences,
+        'keywords': keywords,
+        'summary': summary
+    }
+    ```
+
+**Важно**: 
+* `sentences` - последовательность экземпляров класс `Sentence`, у каждого из которых заполнены поля `_text`, `_preprocessed`, `_encoded`
+* `keywords` - последовательность из 100 ключевых слов для текста, извлечённых с помощью `TFIDFAdapter`
+* `summary` - краткое содержание текста, состоящее из 5 предложений, извлечённых с помощью `TextRankSummarizer`
+
+Интерфейс:
+```py
+def add_text_to_database(self, path_to_text: str) -> None:
+    pass
+```
+
+**Важно №2**: этот метод должен быть вызван для **каждого** текста из списка `paths_to_texts` при создании экземпляра класс `Buddy`.
+
+#### Шаг 9.2 Нахождение текстов, похожих на ключевые слова
+
+Как было описано в Шаге №7, мы можем рассматривать запрос пользователя как текст и, соответственно, выделять из него ключевые слова.
```

Comment:
simpler - we do not extract keywords from query, but we clean it from stop words and each word left is now considered as a key word